src/main/java/com/pinterest/secor/reader/SecorKafkaMessageIterator.java

```diff
@@ -0,0 +1,166 @@
+
+        String dualCommitEnabled = config.getDualCommitEnabled();
+        String offsetStorage = config.getOffsetsStorage();
+        final boolean skipZookeeperOffsetSeek = offsetStorage.equals("kafka") && dualCommitEnabled.equals("true");
+
+        mZookeeperConnector = new ZookeeperConnector(config);
+        mRecordsBatch = new ArrayDeque<>();
+        mKafkaConsumer = new KafkaConsumer<>(props);
+        ConsumerRebalanceListener reBalanceListener = new ConsumerRebalanceListener() {
+            @Override
+            public void onPartitionsRevoked(Collection<TopicPartition> assignedPartitions) {
+                for (TopicPartition topicPartition : assignedPartitions) {
+                    LOG.debug("re-balance will happen for assigned topic partition {}", topicPartition);
+                }
+            }
+
+            @Override
+            public void onPartitionsAssigned(Collection<TopicPartition> collection) {
+                if (skipZookeeperOffsetSeek) {
+                    LOG.debug("offset storage set to kafka. Skipping reading offsets from zookeeper");
+                    return;
+                }
+                Map<TopicPartition, Long> committedOffsets = getCommittedOffsets(collection);
+                committedOffsets.forEach(((topicPartition, offset) -> {
+                    if (offset == -1) {
+                        if (offsetResetConfig.equals("earliest")) {
+                            mKafkaConsumer.seekToBeginning(Collections.singleton(topicPartition));
+                        } else if (offsetResetConfig.equals("latest")) {
+                            mKafkaConsumer.seekToEnd(Collections.singleton(topicPartition));
+                        }
+                    } else {
+                        LOG.debug("Seeking {} to offset {}", topicPartition, Math.max(0, offset));
+                        mKafkaConsumer.seek(topicPartition, Math.max(0, offset));
+                    }
+                }));
+            }
+        };
+
+        String[] subscribeList = config.getKafkaTopicList();
+        if (Strings.isNullOrEmpty(subscribeList[0])) {
+            mKafkaConsumer.subscribe(Pattern.compile(config.getKafkaTopicFilter()), reBalanceListener);
+        } else {
+            mKafkaConsumer.subscribe(Arrays.asList(subscribeList), reBalanceListener);
+        }
+    }
+
+    private Map<TopicPartition, Long> getCommittedOffsets(Collection<TopicPartition> assignment) {
+        Map<TopicPartition, Long> committedOffsets = new HashMap<>();
+
+        for (TopicPartition topicPartition : assignment) {
+            com.pinterest.secor.common.TopicPartition secorTopicPartition =
+                    new com.pinterest.secor.common.TopicPartition(topicPartition.topic(), topicPartition.partition());
+            try {
+                long committedOffset = mZookeeperConnector.getCommittedOffsetCount(secorTopicPartition);
```

Comment:
Does this code need to change if the user is not using ZK to store kafka offsets?

Reply:
No, because secor skips calling this code if `dual.commit.enabled=true` and `offset.storage=kafka` https://github.com/pinterest/secor/pull/450/files#diff-c2752b6d82af124c79e96142788c189dR104 . `dual.commit.enabled=false` and `offset.storage=kafka` combination was ignored prior to this PR and I've kept the same behavior